### PR TITLE
refactor: switch to custom DA client interface

### PR DESCRIPTION
**Pull request by 0xRampey**
- `<daName>.go` -> `client.go` for consistency

This PR switches to a cleaner interface where blob `ID` is not serialised to `[]byte` but left as `any` so that it can take any form or shape. Serialization is left to the `Client` implementation

### CODE REVIEW

File: eigenda/eigenda.go
Comment (aashutoshrathi) [inline eigenda/eigenda.go:1]: rename this to `client.go` too.

File: celestiada/client.go [hunk 1]
```diff
@@ -18,19 +18,19 @@ import (
...
+type Client struct {
 	client    *rpc.Client
 	Namespace share.Namespace
 	gasPrice  float64
 	ctx       context.Context
 }
 
 // Returns an intialised Celestia DA client
-func New(ctx context.Context, lightClientRPCUrl string, authToken string, hexNamespace string, gasPrice float64) (*DAClient, error) {
+func NewClient(ctx context.Context, lightClientRPCUrl string, authToken string, hexNamespace string, gasPrice float64) (*Client, error) {
```
Comment (aashutoshrathi) [inline celestiada/client.go:33]: can we keep this `New` only?

**Review by aashutoshrathi — COMMENTED** [review]
Also, why do every module is suffixed with `da` can we not just add `avail` etc?

**Review by aashutoshrathi — COMMENTED** [review]
Also, I think with this PR we can remove `rollkit/go-da` from `mod` and README and credit them in the `interface.go` as comment

**Comment by 0xRampey** [comment]
> Also, I think with this PR we can remove rollkit/go-da from mod and README 
We still use their `celestia-da` implementation too though so prolly best to leave it as is.

**Pull request merged**